src/server/game/AuctionHouse/AuctionHouseMgr.h

```diff
@@ -28,9 +28,13 @@
 
 class Item;
 class Player;
+class AuctionHouseSearcher;
 
 #define MIN_AUCTION_TIME (12*HOUR)
 #define MAX_AUCTION_ITEMS 160
+#define MAX_AUCTIONS_PER_PAGE 50
+#define AUCTION_SEARCH_DELAY 300
+#define MAX_GETALL_RETURN 55000
```

Comment:
Is there any particular reasoning for this value in particular?

Reply:
Yes, the max allowable single packet size in 3.3.5 client protocol is 0x7FFFFF. A single BuildAuctionInfo structure has a size of 148 bytes. 148 * 55000 = 8140000 which gives us just under the max size of 8388607 with a little bit of margin. The 55k is just a nice arbitrary number close to the limit. There has been previous research to confirm the "getAll" feature did not fully work correctly on high population live realms during WoTLK with lots of auctions but I don't have an exact source at the moment as this was long ago.

Reply:
Gotcha, thank you for the explanation. ❤️

Reply:
Probably worth adding this as a comment to the define value? :)

Reply:
Did you add the comment already?

Reply:
> Did you add the comment already? I would appreciate it

Oops, I read this comment while I was originally at work and forgot about it. Will add now.